tests/equisolve_tests/utils/test_split_data.py

```diff
@@ -0,0 +1,834 @@
+        axis, names, n_groups = "samples", "samples", 3
+        split_tensors_1, actual_grouped_labels_1 = split_data(
+            tensors=test_tensor_map,
+            axis=axis,
+            names=names,
+            n_groups=n_groups,
+            group_sizes=[1, 2, 4],  # adds up to 7
+        )
+        split_tensors_2, actual_grouped_labels_2 = split_data(
+            tensors=test_tensor_map,
+            axis=axis,
+            names=names,
+            n_groups=n_groups,
+            group_sizes=[1 / 8, 2 / 8, 4 / 8],  # adds up to 7/8
+        )
+        assert_equal(target_grouped_labels, actual_grouped_labels_1)
+        assert_equal(actual_grouped_labels_1, actual_grouped_labels_2)
+        for i, j in zip(split_tensors_1[0], split_tensors_2[0]):
+            assert equistore.equal(i, j)
+
+    def test_split_data_unequal_groups(self, test_tensor_map, request):
+        """
+        Tests that splitting data into equal groups using only the `n_groups`
+        splits as expected when the number of unique indices is not divisible
+        by `n_groups`. In this case, there are 8 unique structure idxs in the
+        test tensor maps,
+        """
+        # Get the parameterized TensorMap
+        test_tensor_map = request.getfixturevalue(test_tensor_map)
+        # There are 8 unique structure idxs in total: [0, 1, 2, 3, 4, 5, 6, 8]
+        # Split only the first 7 of these
+        target_grouped_labels = [
+            Labels(
+                names=["samples"],
+                values=np.array([[0]]).reshape(-1, 1),
+            ),
+            Labels(
+                names=["samples"],
+                values=np.array([[1, 2]]).reshape(-1, 1),
+            ),
+            Labels(
+                names=["samples"],
+                values=np.array([[3, 4, 5, 6]]).reshape(-1, 1),
+            ),
+        ]
+        # Split the data using abs and rel group sizes
+        axis, names, n_groups = "samples", "samples", 3
+        split_tensors_1, actual_grouped_labels_1 = split_data(
+            tensors=test_tensor_map,
+            axis=axis,
+            names=names,
+            n_groups=n_groups,
+            group_sizes=[1, 2, 4],  # adds up to 7
+        )
+        split_tensors_2, actual_grouped_labels_2 = split_data(
+            tensors=test_tensor_map,
+            axis=axis,
+            names=names,
+            n_groups=n_groups,
+            group_sizes=[1 / 8, 2 / 8, 4 / 8],  # adds up to 7/8
+        )
+        assert_equal(target_grouped_labels, actual_grouped_labels_1)
+        assert_equal(actual_grouped_labels_1, actual_grouped_labels_2)
+        for i, j in zip(split_tensors_1[0], split_tensors_2[0]):
+            assert equistore.equal(i, j)
+
+
+class TestSplitDataErrors:
+    def test_split_data_errors_arg_tensors(self, test_tensor_map_a):
+        """
+        Checks for exceptions on the `tensors` arg.
+        """
+        # Not passing a TensorMap
+        with pytest.raises(TypeError) as e:
```

Comment:
You may use the `match` parameter here. This saves a couple of line because you do not have to do an extra assert as you done below.